src/Bicep.Core.IntegrationTests/DeployTimeConstantTests.cs

```diff
@@ -126,5 +159,218 @@ public void DtcValidation_RuntimeValueAsDtcPropertyKey_ProducesDiagnostics()
                 ("BCP120", DiagnosticLevel.Error, "This expression is being used in an assignment to the \"tags\" property of the \"Microsoft.Web/serverfarms\" type, which requires a value that can be calculated at the start of the deployment. Properties of dnsZone which can be calculated at the start include \"apiVersion\", \"id\", \"name\", \"type\"."),
             });
         }
+
+        [DataTestMethod]
+        [DataRow(".id")]
+        [DataRow("['id']")]
+        [DataRow("[idAccessor]")]
+        [DataRow("[idAccessor2]")]
+        [DataRow("['${'id'}']")]
+        [DataRow("[idAccessorInterpolated]")]
+        [DataRow("[idAccessorMixed]")]
+        [DataRow("[strArray[0]]")]
+        [DataRow("[first(strArray)]")]
+        [DataRow("[cond ? 'id' : 'name']")]
+        [DataRow("[cond ? first(strArray) : strArray[0]]")]
+        public void DtcValidation_VarForBodyExpression_Ok(string okAccessExp)
+        {
+            StringBuilder textSb = new(GetDtcValidationResourceBaseline());
+            textSb.Append(
+                @"
+param cond bool = false
+var zeroIndex = 0
+var otherIndex = zeroIndex + 2
+var idAccessor = 'id'
+var dStr = 'd'
+var idAccessor2 = idAccessor
+var idAccessorInterpolated = '${idAccessor}'
+var idAccessorMixed = 'i${dStr}'
+var strArray = ['id', 'properties']
+"
+            );
+            textSb.AppendLine(
+                $@"
+var indirect = {{
+  prop: foo{okAccessExp}
+}}
+"
+            );
+
+            var okCase = 0;
+
+            void AddForBodyExpressionVariants(string valueExp)
+            {
+                textSb.AppendLine($"var ok{++okCase} = [for i in range(0, 2): {valueExp}]");
+                textSb.AppendLine(
+                    $@"var ok{++okCase} = [for i in range(0, 2): {{
+  prop: {valueExp}
+}}]"
+                );
+                textSb.AppendLine(
+                    $@"var ok{++okCase} = [for i in range(0, 2): {{
+  '${{{valueExp}}}': 'value'
+}}]"
+                );
+            }
+
+            AddForBodyExpressionVariants($"foo{okAccessExp}");
+            AddForBodyExpressionVariants($"existingFoo{okAccessExp}");
+            AddForBodyExpressionVariants("indirect.prop");
+            textSb.AppendLine($@"var ok{++okCase} = [for i in range(0, 2): foo::fooChild{okAccessExp}]");
+
+            var arrayAccessorExps = new[] { "0", "i", "i + 2", "zeroIndex", "otherIndex" };
+            foreach (var arrAccessorExp in arrayAccessorExps)
+            {
+                AddForBodyExpressionVariants($"foos[{arrAccessorExp}]{okAccessExp}");
+            }
```

Comment:
I agree it's not easily readable. I did try expanding out all the test cases from an early version of this scenario test but because it's specifically testing "var for-bodies" cases and creating a variable per permutation, I hit the variable limit of 256 when I ran the baseline test. Splitting the entire test by using the access expressions as `DataRow` solved this.

I think one benefit to this approach is the ability to run only a single `DataRow` test and then debug with conditional breakpoints, although it's still not easy in some cases.

Is there a way to bypass the variable limit for a baseline, or would multiple baselines be ok?

Reply:
The variable limit is [an ARM runtime limit](https://learn.microsoft.com/en-us/azure/azure-resource-manager/templates/best-practices#template-limits), so there's no way to disable it. 

I think if you're testing >256 unique behaviors, you'll need more than one test. It doesn't look like that's the case here (e.g., the accessor type in each of `<resource>.id`, `<resource>['id']`, and `<resource>['${'id'}']` is `'id'`, even if there's an intervening variable reference), so I think the limit may be being breached because the same behavior is being exercised in ways that `DeployTimeConstant(In)directViolationVisitor` doesn't handle as distinct cases.

Reply:
Split into 2 baselines, valid and invalid. Expanded out the cases as object properties to work around the variable limit.